openquake/smt/residuals/residual_plotter.py

```diff
@@ -813,96 +813,12 @@ def PlotLoglikelihoodWithSpectralPeriod(residuals,filename,custom_cycler=0,
-         if residuals.imts[imt_idx]=='Ia':
-             imt_append=imt_append.drop(imt_append.loc['Ia'])
-        
-    imt_append_list=pd.DataFrame()
-    for idx in range(0,len(imt_append)):
-         imt_append_list[idx]=imt_append.iloc[idx]
-    imt_append=imt_append.reset_index()
-    imt_append_list.columns=imt_append.imt
-    residuals.imts=list(imt_append_list)
-    
-    # Convert imt_list to array
-    x_model_weights=pd.DataFrame([imt2tup(imts) for imts in residuals.imts],
-                                 columns=['imt_str','imt_float'])
-    for imt_idx in range(0,np.size(residuals.imts)):
-         if x_model_weights.imt_str[imt_idx]=='PGA':
-            x_model_weights.imt_float.iloc[imt_idx]=0
-    x_model_weights=x_model_weights.dropna() # Remove any non-acceleration im
-    
-    # Generate attributes required from residuals for table
-    residuals.get_loglikelihood_values(residuals.imts)
-
-    # Produce series of residuals.gmpe_list used to index model_weights_with_imt
-    gmpe_list_series=pd.Series(pd.DataFrame(residuals.gmpe_list,index=
-                                            np.arange(0,len(residuals.gmpe_list)
-                                                      ,1)).columns)
-    
-    # Define colours for plots
-    colour_cycler = (cycler(color=['r', 'g', 'b', 'y','lime','k','dodgerblue',
-                                   'gold','0.8','mediumseagreen','0.5',
-                                   'tab:orange', 'tab:purple','tab:brown',
-                                   'tab:pink'])*cycler(linestyle=['-']))
-    if type(custom_cycler)==type(cycler(colour='b')):
-       colour_cycler=custom_cycler
-       
-
-    # Plot model weights w.r.t. spectral period
-    model_weights_with_imt=pd.DataFrame(residuals.model_weights_with_imt)
-    fig_model_weights, ax_model_weights = plt.subplots(figsize=(10, 8))
-    ax_model_weights.set_prop_cycle(colour_cycler)
-    for gmpe in range(0,len(gmpe_list_series)):
-        y_model_weights=np.array(model_weights_with_imt.iloc[gmpe])
-        ax_model_weights.scatter(x_model_weights.imt_float,
-                                 y_model_weights)
-        tmp = str(residuals.gmpe_list[gmpe_list_series[gmpe]])
-        ax_model_weights.plot(x_model_weights.imt_float,y_model_weights,
-                              label=tmp.split('(')[0])
-    ax_model_weights.set_xlabel('Spectral Period (s)')
-    ax_model_weights.set_ylabel('Model Weight')
-    ax_model_weights.set_title(
-        'Sample loglikelihood based model weights (Scherbaum et al. 2009)',fontsize='16')
-    ax_model_weights.legend(loc='upper right',ncol=2,fontsize='x-small')
-    _save_image(filename, plt.gcf(), filetype, dpi)
-    
-    # Reassign original imts to residuals.imts
-    residuals.imts = preserve_imts
-    
-def PlotEDRWithSpectralPeriod(residuals,filename,custom_cycler=0,
+def PlotEDRMetricsWithSpectralPeriod(residuals,filename,custom_cycler=0,
```

Comment:
the name of a function must start with lower case letter. If you want to comply with pep8 standard you should also leave a space after each comma)